src/librustc_typeck/check/method/probe.rs

```diff
@@ -895,20 +895,46 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         let trait_substs = self.fresh_item_substs(trait_def_id);
         let trait_ref = ty::TraitRef::new(trait_def_id, trait_substs);
 
-        for item in self.impl_or_trait_item(trait_def_id) {
-            // Check whether `trait_def_id` defines a method with suitable name:
-            if !self.has_applicable_self(&item) {
-                debug!("method has inapplicable self");
-                self.record_static_candidate(TraitSource(trait_def_id));
-                continue;
+        // If a trait alias, elaborate its bounds magically...
+        // TODO? figure out a better way to determine if this is an alias
```

Comment:
Try using the `tcx.is_trait_alias` method instead, which I wrote in a previous PR. It works cross-crate, which this doesn't.